aggregator/src/aggregator/http_handlers.rs

```diff
@@ -44,50 +44,58 @@ impl Handler for Error {
         match self {
             Error::InvalidConfiguration(_) => conn.with_status(Status::InternalServerError),
             Error::MessageDecode(_) => {
-                conn.with_problem_details(DapProblemType::InvalidMessage, None)
-            }
-            Error::ReportRejected(task_id, _, _) => {
-                conn.with_problem_details(DapProblemType::ReportRejected, Some(task_id))
+                conn.with_problem_details(DapProblemType::InvalidMessage, None, None)
             }
+            Error::ReportRejected(task_id, _, _, reason) => conn.with_problem_details(
+                DapProblemType::ReportRejected,
+                Some(task_id),
+                Some(reason.detail()),
+            ),
             Error::InvalidMessage(task_id, _) => {
-                conn.with_problem_details(DapProblemType::InvalidMessage, task_id.as_ref())
+                conn.with_problem_details(DapProblemType::InvalidMessage, task_id.as_ref(), None)
```

Comment:
nit: I don't love having the majority of these callsites take `None`. Maybe we could have a problem details builder where you would optionally chain a call to `.with_detail(&str)`? Or you could even specialize it further for `ReportRejectedReason`.

Reply:
WDYT about https://github.com/divviup/janus/pull/2311/commits/1b6d22e2398f86d40dd94cb707de9898ae1d5f96? Not much of a LOC savings but it is nice having named arguments.

Reply:
Much better! I like to see a bunch of `None` arguments disappear.